airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -868,10 +784,131 @@ private List<MetricAttribute> generateMetricAttributes(final ConnectionUpdaterIn
     return metricAttributes;
   }
 
+  private void initInternalState(final ConnectionUpdaterInput connectionUpdaterInput) {
+    connectionId = connectionUpdaterInput.getConnectionId();
+
+    // workflow state is only ever set in test cases. for production cases, it will always be null.
+    if (connectionUpdaterInput.getWorkflowState() != null) {
+      workflowState = connectionUpdaterInput.getWorkflowState();
+    }
+
+    if (connectionUpdaterInput.isSkipScheduling()) {
+      workflowState.setSkipScheduling(true);
+    }
+  }
+
+  private void waitUntilNextRun(final ConnectionUpdaterInput connectionUpdaterInput) {
+    final Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+
+    Workflow.await(timeToWait,
+        () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+    workflowState.setDoneWaiting(true);
+  }
+
+  private boolean needToRun(final ConnectionUpdaterInput connectionUpdaterInput) {
```

Comment:
Could this be replaced by the `handleTermination` function I suggested below?

Reply:
Not really because we don't wan't update the status here since it will be done once we exited the cancellation scope so the status is updated in a single place. I will remove the continueAsNew from it.